### PR TITLE
Add graceful shutdown

### PR DESCRIPTION
**Pull request by wanghaowei0107**
#868

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -224,4 +226,10 @@ func main() {
 	} else {
 		log.Fatal(http.ListenAndServe(*listenAddress, exp))
 	}
+
+	s := make(chan os.Signal, 1)
+	signal.Notify(s, syscall.SIGINT, syscall.SIGTERM)
+	_sig := <-s
```
Comment (oliver006) [inline main.go:232]: Thanks for the PR!

How does this fix the issue? 
The listen call in line 225/227 should block so I don't think we ever get here.

**Comment by oliver006** [comment]
Thanks for looking into this!
How does this fix the issue? 
The listen call in line 225/227 should block so I don't think we ever down to where you catch the signal.

**Comment by wanghaowei0107** [comment]
@oliver006 Hello! Thanks for your reply! I rewrote the code, but the workflow is pending for a long time. Can you help me find out what's going on?

**Comment by wanghaowei0107** [comment]
test

**Comment by oliver006** [comment]
Thanks for updating this PR, this looks a lot better. 
Did you have a chance to test it? Does it work?

**Comment by wanghaowei0107** [comment]
Yes,I did test in local.It's work.
By the way, Can you tell me why drone/pr check is always pending on first step? I don't know how to pass this check. Thank you.
微信图片_20240115101136

**Review by oliver006 — APPROVED** [review]
Sorry for the delay.
This looks good - please one last change for the message that is printed and then we should be good to go.

**Review by oliver006 — APPROVED** [review]
Looks good - thanks for the PR!

**Comment by wanghaowei0107** [comment]
So when will this feature be released?

**Comment by oliver006** [comment]
Released as https://github.com/oliver006/redis_exporter/releases/tag/v1.57.0

**Pull request merged**